databuilder/tests/unit/extractor/test_eventbridge_extractor.py

```diff
@@ -0,0 +1,294 @@
+            "type": "object",
+            "properties": {
+                "id": {"type": "string"},
+                "status": {"type": "string"},
+                "customer": {"$ref": "#/definitions/Customer"},
+            },
+            "required": ["id", "status", "customer"],
+        },
+        "Customer": {
+            "type": "object",
+            "properties": {"id": {"type": "string"}, "name": {"type": "string"}},
+            "required": ["id", "name"],
+        },
+    },
+    "properties": {
+        "version": {
+            "$id": "#/properties/version",
+            "type": "string",
+            "description": "version description",
+        },
+        "id": {"$id": "#/properties/id", "type": "string",},
+        "detail-type": {"$id": "#/properties/detail-type", "type": "string",},
+        "source": {"$id": "#/properties/source", "type": "string",},
+        "account": {"$id": "#/properties/account", "type": "string",},
+        "time": {"$id": "#/properties/time", "type": "string",},
+        "region": {"$id": "#/properties/region", "type": "string",},
+        "resources": {
+            "$id": "#/properties/resources",
+            "type": "array",
+            "additionalItems": True,
+            "items": {"$id": "#/properties/resources/items", "type": "string"},
+        },
+        "detail": {"$ref": "#/definitions/BookingDone"},
+    },
+}
+
+json_draft_4_customer_type = f"struct<id:string,name:string>"
+json_draft_4_booking_type = (
+    f"struct<id:string,status:string,customer:{json_draft_4_customer_type}>"
+)
+json_draft_4_booking_done_type = f"struct<booking:{json_draft_4_booking_type}>"
+
+# patch whole class to avoid actually calling for boto3.client during tests
+@patch("databuilder.extractor.eventbridge_extractor.boto3.client", lambda x: None)
+class TestEventBridgeExtractor(unittest.TestCase):
+    def setUp(self) -> None:
+        logging.basicConfig(level=logging.INFO)
+
+        self.conf = ConfigFactory.from_dict(
+            {EventBridgeExtractor.REGISTRY_NAME: registry_name}
+        )
+        self.maxDiff = None
+
+    def test_extraction_with_empty_query_result(self) -> None:
+        """
+        Test Extraction with empty result from query
+        """
+        with patch.object(EventBridgeExtractor, "_search_schemas"):
+            extractor = EventBridgeExtractor()
+            extractor.init(self.conf)
+
+            results = extractor.extract()
+            self.assertEqual(results, None)
+
+    def test_extraction_with_single_result_openapi_3(self) -> None:
+        with patch.object(EventBridgeExtractor, "_search_schemas") as mock_search:
+            mock_search.return_value = [{"Content": json.dumps(test_schema_openapi_3),}]
+
+            extractor = EventBridgeExtractor()
+            extractor.init(self.conf)
+
+            expected = TableMetadata(
+                "eventbridge",
+                "gold",
+                test_schema_openapi_3["info"]["title"],
```

Comment:
From looking into the responses, it seems that this always results in `Event`.
Could we instead use the schema name? This is important, as this is how it will show up in Amundsen.

Reply:
That's not necessarily true, you can anything there. I think some of the default AWS schemas do this too.